Auth:Password Reset

neetkiss opened the pull request:
Güncelleme;

- [ x ] Password Token Store
- [ x ] Password Token Find
- [ x ] Password Reset
- [ x ] ResetPassword Model
- [ x ] Forgot Password
- [ x ] Mail Notfiy
- [ x ] Testler
- [ x ] Api Route
@sustartx hocam ekler burada yer almaktadır.

sustartx committed:
Merge pull request #21 from acikkaynak/dev

Birkaç sorun giderildi.

Commit:
Güncelleme;

- [ x ] Password Token Store
- [ x ] Password Token Find
- [ x ] Password Reset
- [ x ] ResetPassword Model
- [ x ] Forgot Password
- [ x ] Mail Notfiy
- [ x ] Testler
- [ x ] Api Route

sustartx commented on app/Http/Controllers/AuthController.php:
```diff
@@ -125,4 +128,232 @@ public function guard($guard = 'web')
+     *
+     */
+    public function password_reset(Request $request) {
+        $rules = [
```
Validation işlemini Request sınıfı içinde yapabiliriz.

sustartx commented on app/Http/Controllers/AuthController.php:
```diff
@@ -125,4 +128,232 @@ public function guard($guard = 'web')
+        $rules = [
+            'email' => 'required|string|email|exists:users',
+            'token' => 'required|string',
+            'password' => 'required|string|min:6|confirmed',
```
Password alanları için min:6 yerine min:8 yapalım.

sustartx commented on app/Http/Controllers/AuthController.php:
```diff
@@ -125,4 +128,232 @@ public function guard($guard = 'web')
+     *
+     */
+    public function password_store_token(Request $request) {
+        $rules = [
```
Validation işlemini Request içinde yapalım.

sustartx commented on app/Http/Controllers/AuthController.php:
```diff
@@ -125,4 +128,232 @@ public function guard($guard = 'web')
+            'password_confirmation' => 'required|string|min:6',
+        ];
+        $validator = Validator::make($request->all(), $rules);
+        if ($validator->fails()) {
```
Validation response kalıbı bu haliyle değişiyor, bunun yerine Request yazarak aynı kalıpta dönmesini sağlayabiliriz.

sustartx commented on app/Http/Controllers/AuthController.php:
```diff
@@ -125,4 +128,232 @@ public function guard($guard = 'web')
+     *
+     */
+    public function profile_password_reset(Request $request) {
+        $rules = [
```
Validation işlemini Request içine alalım.

sustartx replied:
Password min:6 yerine min:8 olmalı.

Commit:
Api ve AuthController Düzenlendi.
1)Api.php daha anlaşılır bir şekilde biçimlendirildi.
2)AuthController içerisinde düzenlemeler yapıldı.
   -- Classlar daha net anlaşılır şekilde isimlendirildi.
   -- $request->object kullanımı $data[''] olarak güncellendi. validate kodları içerisinde kullanıldı.
   -- Token Generate işlemi her 180 saniyede bir yapılacak şekilde yeniden güncellendi.
   -- Responselar yeniden düzenlendi.
   --Şifre güncelleme kısmı yeniden düzenlendi.
   --Profil için şifre güncelleme kısmı yeniden düzenlendi.

Commit:
Api ve AuthController Düzenlendi.
1)AuthController
 -- Açıklamalar kaldırıldı.
 -- Requestler temize çekildi.
 -- 151. Satırda ki Carbon kullanımı diffInMinutes ile doğrudan çalıştırıldı.
 -- Response ile dönen mesajlar yeniden düzenlendi.
 -- 164. satırda ki resetPassword bölümünde ki ekleme kısmında tarih eklenmemişti. created_at sütunu için eklendi.
 -- passwordTokenCheck fonksiyonunda 198.satırda ki if yapısı düzenlendi. $token olan kısım $resetpassword olarak güncellendi.
 -- $data ve $request->object olan tüm kısımlar request ile bağlanıp $request->get('') olarak güncellendi.
 -- Auth::User()-> olan kısımlar direktifler doğrultusunda Auth::user() olarak yeniden güncellendi.(örnek:283. satırda mevcut)
2)ResetPassword modeli
 --  İçerisinde fillable olarak created_at eklendi.
4) Request dosyaları oluşturuldu.
 --ResetWithProfileRequest
 --ResetWithTokenRequest
 --TokenGenerateRequest

Commit:
AuthController tespit ettiğim hatalar
1)187. satır status => true olan kısım false olarak değiştirildi.
2)299. satırda ki yazım ve tanım hataları giderildi.
